packages/x-internals/src/rafThrottle/rafThrottle.test.ts

```diff
@@ -0,0 +1,231 @@
+    expect(fn).not.toHaveBeenCalled();
+
+    // Execute pending RAF callback
+    vi.advanceTimersToNextFrame();
+
+    // Function should be called once with the last arguments
+    expect(fn).toHaveBeenCalledTimes(1);
+    expect(fn).toHaveBeenCalledWith(3);
+  });
+
+  it('should preserve the last arguments when called multiple times', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    throttled('first', 1);
+    throttled('second', 2);
+    throttled('third', 3);
+
+    vi.advanceTimersToNextFrame();
+
+    expect(fn).toHaveBeenCalledTimes(1);
+    expect(fn).toHaveBeenCalledWith('third', 3);
+  });
+
+  it('should cancel pending calls when clear is called', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    throttled(1);
+    throttled(2);
+    throttled(3);
+
+    // Clear before RAF executes
+    throttled.clear();
+
+    vi.advanceTimersToNextFrame();
+
+    // Function should not be called
+    expect(fn).not.toHaveBeenCalled();
+  });
+
+  it('should allow new calls after clear', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    throttled(1);
+    throttled.clear();
+    throttled(2);
+
+    vi.advanceTimersToNextFrame();
+
+    expect(fn).toHaveBeenCalledTimes(1);
+    expect(fn).toHaveBeenCalledWith(2);
+  });
+
+  it('should execute on every frame when called continuously', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    // First frame
+    throttled(1);
+    vi.advanceTimersToNextFrame();
+
+    // Second frame
+    throttled(2);
+    vi.advanceTimersToNextFrame();
+
+    // Third frame
+    throttled(3);
+    vi.advanceTimersToNextFrame();
+
+    // Should execute once per frame
+    expect(fn).toHaveBeenCalledTimes(3);
+    expect(fn).toHaveBeenNthCalledWith(1, 1);
+    expect(fn).toHaveBeenNthCalledWith(2, 2);
+    expect(fn).toHaveBeenNthCalledWith(3, 3);
+  });
+
+  it('should batch multiple calls within the same frame', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    // Simulate rapid calls within the same frame
+    for (let i = 0; i < 100; i += 1) {
+      throttled(i);
+    }
+
+    vi.advanceTimersToNextFrame();
+
+    // Should only execute once with the last value
+    expect(fn).toHaveBeenCalledTimes(1);
+    expect(fn).toHaveBeenCalledWith(99);
+  });
+
+  it('should work with functions that have no arguments', async () => {
+    const fn = vi.fn();
+    const throttled = rafThrottle(fn);
+
+    throttled();
+    throttled();
+    throttled();
+
+    vi.advanceTimersToNextFrame();
+
+    expect(fn).toHaveBeenCalledTimes(1);
+  });
+
+  it('should work with functions that return values', async () => {
+    const fn = vi.fn().mockReturnValue(42);
+    const throttled = rafThrottle(fn);
+
+    throttled(1);
+
+    vi.advanceTimersToNextFrame();
+
+    expect(fn).toHaveBeenCalledWith(1);
+    expect(fn).toHaveReturnedWith(42);
+  });
```

Comment:
Are these adding anything? Why wouldn't they work in these cases?

Especially in the second case we're not asserting that `throttled(1)` returns anything, just the that mocked function returns what we mocked it to return in the test. It seems unnecessary.

Reply:
The return one is valid, values should be properly returned, the first one is a bit useless so I removed it.